FormalConjectures/ErdosProblems/1056.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 1056
+
+*Reference:* [erdosproblems.com/1056](https://www.erdosproblems.com/1056)
+-/
+
+namespace Erdos1056
+
+/--
+Checks if the modular product of each interval equals 1 modulo the prime p,
+where intervals are defined by consecutive boundaries.
+-/
+def Erdos1056For (p : ℕ) (boundaries : List ℕ) : Prop :=
```

Comment:
I would define boundaries as a `StrictMono` from `Fin (k + 1)` for some `k` as before, rather than a list. I think it will be slightly more convenient to prove things in this setting (for example I think doing induction will be easier).